### DISCUSSION TITLE
populating virtual with same name as existing path

### DISCUSSION THREAD

**Discussion by vineetdigit** [Automattic/mongoose#10765] [Q&A] [answered]
Hello,

I have a schema that contains a path "authoredBy" that has type Number.

```
const SomeSchema = new Schema(
  {
    authoredBy: {
      type: Number,
      required: true,
    },

    .....
  });

```
I want this path populated with a document from another collection, and I'm defining a virtual as follows
```
SomeSchema.virtual("authoredBy", {
  localField: "authoredBy",
  foreignField: "id", //  find user whose "id" == "authoredBy"
  justOne: true,
});
```
This results in error message as
Error: Virtual path "authoredBy" conflicts with a real path in the schema

If I change the virtual name to something else like "authorInfo", then it works. But then I have two paths in the populated document (authoredBy and authorInfo) which is acceptable, but not ideal. I want to have same experience as conventionally populating paths that have type ObjectId in which the pathname is simply populated with a document from another collection. 

Is this possible with virtual population?

**Comment by vkarpov15** [comment, marked as answer]
Unfortunately not with virtual populate. There's no way for a path to be both a real path and a virtual in a schema. 

Why not just use conventional populate?

```javascript
const SomeSchema = new Schema(
  {
    authoredBy: {
      type: Number,
      required: true,
      ref: 'Author'
    },

    .....
  });
```

**Reply by vineetdigit** [reply]
Thanks for your reply @vkarpov15 

I was under the impression that conventional populate works only for paths that have the type of ObjectId, but not Number. That's why I haven't tried it yet. Should conventional populate with authoredBy being a Number?

**Reply by vkarpov15** [reply]
Yes, conventional populate should work fine even if `authoredBy` is a number.
